internal/cli/release_list.go

```diff
@@ -0,0 +1,391 @@
+				details = append(details, "user:"+user)
+			} else if b.Preload.Build != nil {
+				build := b.Preload.Build
+				// labels have been set, safe to use them
+
+				if user, ok := build.Labels["common/user"]; ok {
+					details = append(details, "build-user:"+user)
+				}
+				if bp, ok := build.Labels["common/languages"]; ok {
+					details = append(details, niceLanguages(bp))
+				}
+
+				if img, ok := build.Labels["common/image-id"]; ok {
+					img = shortImg(img)
+
+					details = append(details, "image:"+img)
+				}
+			}
+
+			if b.Preload.Artifact != nil {
+				artDetails := fmt.Sprintf("artifact:%s", c.flagId.FormatId(b.Preload.Artifact.Sequence, b.Preload.Artifact.Id))
+				if len(details) == 0 {
+					details = append(details, artDetails)
+				} else if c.flagVerbose && b.Preload.Build != nil {
+					details = append(details,
+						artDetails,
+						fmt.Sprintf("build:%s", c.flagId.FormatId(b.Preload.Build.Sequence, b.Preload.Build.Id)))
+				}
+			}
+
+			if c.flagVerbose {
+				for k, val := range b.Labels {
+					if strings.HasPrefix(k, "waypoint/") {
+						continue
+					}
+
+					if len(val) > 30 {
+						val = val[:30] + "..."
+					}
+
+					extraDetails = append(extraDetails, fmt.Sprintf("Release.%s:%s", k, val))
+				}
+				sort.Strings(extraDetails)
+			}
+
+			sort.Strings(details)
+			var firstDetails string
+			if len(details) > 0 {
+				firstDetails = details[0]
+			}
+
+			var columns []string
+
+			columns = []string{
+				status,
+				c.flagId.FormatId(b.Sequence, b.Id),
+				c.flagId.FormatId(b.Preload.Deployment.Sequence, b.Id),
+				b.Component.Name,
+				firstDetails,
+				startTime,
+				completeTime,
+				statusReportComplete,
+			}
+
+			if c.flagUrl {
+				url := "n/a"
+				if b.Url != "" {
+					url = b.Url
+				} else if releaseBundle.Release.Url != "" {
+					url = releaseBundle.Release.Url
+				}
+				columns = append(columns, url)
+			}
+
+			if !b.Unimplemented {
+				tbl.Rich(
+					columns,
+					[]string{
+						statusColor,
+					},
+				)
+
+				if len(details) > 1 {
+					for _, dr := range details[1:] {
+						tbl.Rich([]string{"", "", "", dr}, nil)
+					}
+				}
+
+				if len(extraDetails) > 0 {
+					for _, dr := range extraDetails {
+						tbl.Rich([]string{"", "", "", dr}, nil)
+					}
+				}
+			}
+		}
+
+		c.ui.Table(tbl)
+
+		return nil
+	})
+	if err != nil {
+		return 1
+	}
+
+	return 0
+}
+
+func (c *ReleaseListCommand) displayJson(releases []*pb.UI_ReleaseBundle) error {
+	var output []map[string]interface{}
+
+	for _, rel := range releases {
+		i := map[string]interface{}{}
+
+		i["id"] = rel.Release.Sequence
+		i["deploymentId"] = rel.Release.Preload.Deployment.Sequence
+		i["application"] = rel.Release.Application
+		i["workspace"] = rel.Release.Workspace.Workspace
+		i["url"] = rel.Release.Url
+		i["labels"] = rel.Release.Labels
+		i["component"] = rel.Release.Component.Name
+		i["status"] = c.statusJson(rel.Release.Status)
+		i["latestStatusReport"] = rel.LatestStatusReport
```

Comment:
Maybe under this we also include the `Unimplemented` field? We're also omitting any release that didn't actually happen, maybe we should do that for the json view too?

Reply:
Good catch! I modified to have the field for now, but maybe we should omit altogether in JSON too. Since if those are just Waypoint internal releases that the user doesn't care about ever, they'd just add noise to the JSON output. Unless it's important to have _some_ record of it?